api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1673,18 +1673,66 @@ type AWSServiceEndpoint struct {
 }
 
 type AzurePlatformSpec struct {
+	// Credentials is the object containing the Azure credentials needed when creating and managing Azure cloud
+	// infrastructure resources
+	//
+	// +kubebuilder:validation:Required
 	Credentials corev1.LocalObjectReference `json:"credentials"`
-	// The cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+
+	// Cloud is the cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+	//
 	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
```

Comment:
Is the enum in here case sensitive?

Reply:
I believe so - https://github.com/openshift/hypershift/blob/ac66d8f59abacf5e2650e0bc602c1cc8fc1bde79/vendor/github.com/openshift/api/config/v1/types_infrastructure.go#L584

Reply:
bit confusing because the link above has them all Capitalized but then the enum is comparing them to camel casing

Reply:
I updated it a bit to include the other two enum options mentioned in: https://github.com/openshift/hypershift/blob/ac66d8f59abacf5e2650e0bc602c1cc8fc1bde79/vendor/github.com/openshift/api/config/v1/types_infrastructure.go#L581C4-L581C124

Reply:
its not case sensitive afaik